src/Storefront/Test/Controller/CartLineItemControllerTest.php

```diff
@@ -40,6 +48,41 @@ public function testErrorBehaviourInFlashMessages(): void
         static::assertTrue($response->isRedirect(), $response->getContent());
     }
 
+    /**
+     * @dataProvider productNumbers
+     */
+    public function testAddProductByNumber(string $productId, string $productNumber): void
+    {
+        $this->getFlashBag()->clear();
+        $contextToken = Uuid::randomHex();
+
+        /** @var CartService $cartService */
+        $cartService = $this->getContainer()->get(CartService::class);
+        $this->createProduct($productId, $productNumber);
+        $request = $this->createRequest();
+        $request->attributes->set('number', $productNumber);
+
+        $salesChannelContext = $this->createSalesChannelContext($contextToken);
+        $this->getContainer()->get(CartLineItemController::class)->addProductByNumber($request, $salesChannelContext);
```

Comment:
maybe some assertions on the response?

Reply:
Well... with the data bag and the product being in the cart everything should have worked fine. I'll a check for the statuscode.